impl/src/fmt/mod.rs

```diff
@@ -133,7 +133,69 @@ impl ToTokens for FmtAttribute {
 }
 
 impl FmtAttribute {
-    /// Returns an [`Iterator`] over bounded [`syn::Type`]s and trait names.
+    /// Checks whether this [`FmtAttribute`] can be replaced with a transparent delegation (calling
+    /// a formatting trait directly instead of interpolation syntax).
+    ///
+    /// If such transparent call is possible, the returns an [`Ident`] of the delegated trait and
+    /// the [`Expr`] to pass into the call, otherwise [`None`].
+    ///
+    /// [`Ident`]: struct@syn::Ident
+    fn transparent_call(&self) -> Option<(Expr, syn::Ident)> {
+        // `FmtAttribute` is transparent when:
+
+        // (1) There is exactly one formatting parameter.
+        let lit = self.lit.value();
+        let param =
+            parsing::format(&lit).and_then(|(more, p)| more.is_empty().then_some(p))?;
+
+        // (2) And the formatting parameter doesn't contain any modifiers.
+        if param
+            .spec
+            .map(|s| {
+                s.align.is_some()
+                    || s.sign.is_some()
+                    || s.alternate.is_some()
+                    || s.zero_padding.is_some()
+                    || s.width.is_some()
+                    || s.precision.is_some()
+                    || !s.ty.is_trivial()
+            })
+            .unwrap_or_default()
+        {
+            return None;
+        }
+
+        let expr = match param.arg {
+            // (3) And either exactly one positional argument is specified.
+            Some(parsing::Argument::Integer(_)) | None => (self.args.len() == 1)
+                .then(|| self.args.first())
+                .flatten()
+                .map(|a| a.expr.clone()),
+
+            // (4) Or the formatting parameter's name refers to some outer binding.
+            Some(parsing::Argument::Identifier(name)) if self.args.is_empty() => {
+                Some(format_ident!("{name}").into())
+            }
+
+            // (5) Or exactly one named argument is specified for the formatting parameter's name.
+            Some(parsing::Argument::Identifier(name)) => (self.args.len() == 1)
+                .then(|| self.args.first())
+                .flatten()
+                .filter(|a| a.alias.as_ref().map(|a| a.0 == name).unwrap_or_default())
+                .map(|a| a.expr.clone()),
+        }?;
+
+        let trait_name = param
+            .spec
+            .map(|s| s.ty)
+            .unwrap_or(parsing::Type::Display)
+            .trait_name();
```

Comment:
When is this "default" case applicable?

Reply:
@JelteF when `format_spec` is not specified, i.e. `{}`, or `{ident}`. See [grammar for details](https://doc.rust-lang.org/stable/std/fmt/index.html#syntax).